File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/exceptions/FunctionCompletedExceptionally.java
package io.contract_testing.contractcase.exceptions;

public class FunctionCompletedExceptionally extends RuntimeException {


  private final String errorClassName;
  private final String exceptionMessage;
  private final Object errorInternals;

  public FunctionCompletedExceptionally(String errorClassName, String exceptionMessage) {
    this(errorClassName, exceptionMessage, null);
  }

  public FunctionCompletedExceptionally(String errorClassName, String exceptionMessage,
      Object errorInternals) {
    super(
        "Function completed exceptionally: "
            + errorClassName +
            (exceptionMessage != null
                ? " " + exceptionMessage
                : ""
            )
    );
    this.errorClassName = errorClassName;
    this.exceptionMessage = exceptionMessage;
    this.errorInternals = errorInternals;
  }

  public String getErrorClassName() {
    return errorClassName;
  }

  public String getExceptionMessage() {
    return exceptionMessage;
  }

  /**
   * The serialised content of the exception as described by the contract, if any. This is the
   * deserialised JSON content (typically a Map, List, or primitive), or null if the contract did
   * not describe a errorInternals for this error.
   *
   * @return the errorInternals, or null
   */
  public Object getErrorInternals() {
    return errorInternals;
  }

}

File: packages/contract-case-dsl-js-internals/src/entities/errors/FunctionCompletedExceptionally.ts
/**
 * Thrown by a mock function (during contract definition) when the contract
 * describes the function as throwing an error. Carries the details of the
 * error as described by the contract.
 *
 * @public
 */
export class FunctionCompletedExceptionally extends Error {
  /** The class name of the error, as described by the contract */
  readonly errorClassName: string;

  /** The message of the error as described by the contract, if any */
  readonly exceptionMessage: string | undefined;

  /**
   * The serialised content of the error as described by the contract, if any.
   * This is the example error internals from the contract's errorInternals matcher, or
   * undefined if the contract did not describe error internals for this error.
   */
  readonly errorInternals: unknown;

  constructor(
    errorClassName: string,
    exceptionMessage: string | undefined,
    errorInternals: unknown,
  ) {
    super(
      `Function completed exceptionally: ${errorClassName}${
        exceptionMessage != null ? ` ${exceptionMessage}` : ''
      }`,
    );
    Object.setPrototypeOf(this, new.target.prototype);
    this.name = 'FunctionCompletedExceptionally';
    this.errorClassName = errorClassName;
    this.exceptionMessage = exceptionMessage;
    this.errorInternals = errorInternals;
  }
}
